genienlp/predict.py

```diff
@@ -391,48 +370,71 @@ def prepare_data_iterators(args, val_sets, numericalizer, device):
-            for index, set_ in enumerate(val_set):
-                loader, original_order = make_data_loader(
-                    set_, numericalizer, bs, device, train=False, return_original_order=True
-                )
-                task_iter.append((task, task_languages[index], loader, original_order))
-        # single language task or no separate eval
-        else:
-            loader, original_order = make_data_loader(
-                val_set[0], numericalizer, bs, device, train=False, return_original_order=True
-            )
-            task_iter.append((task, task_languages, loader, original_order))
+        loader, original_order = make_data_loader(val_set, numericalizer, bs, device, train=False, return_original_order=True)
+        task_iter.append((task, loader, original_order))
 
         iters.extend(task_iter)
-        task_index += 1
 
     return iters
 
 
+def create_output_lines(args, index, generation_output):
+    predictions = generation_output.raw_predictions if args.translate_return_raw_outputs else generation_output.predictions
+    if args.one_output_per_line:
+        lines = [
+            '\t'.join(
+                [
+                    generation_output.example_ids[index],
+                    prediction,
+                    generation_output.answers[index],
+                    generation_output.contexts[index],
+                ]
+            )
+            for prediction in predictions[index]
+        ]  # one line per generation output
+    else:
+        lines = [
+            '\t'.join(
+                [
+                    generation_output.example_ids[index],
+                    *predictions[index],
+                    generation_output.answers[index],
+                    generation_output.contexts[index],
+                ]
+            )
+        ]  # one line with all generation outputs separated by '\t'
+    if args.calibrator_paths is not None:
+        for score in generation_output.confidence_scores:
+            lines = [line + '\t' + str(score[index]) for line in lines]  # append score to all lines
+    return lines
+
+
+def get_metrics_to_compute(args, task):
+    metrics_to_compute = task.metrics
+    metrics_to_compute += args.extra_metrics
+    metrics_to_compute = [metric for metric in task.metrics if metric not in ['loss']]
+    if args.main_metric_only:
+        metrics_to_compute = [metrics_to_compute[0]]
+    return metrics_to_compute
+
+
 def run(args, device):
-    # TODO handle multiple languages
-    src_lang = args.pred_src_languages[0]
-    tgt_lang = args.pred_tgt_languages[0]
 
+    # TODO handle multiple languages
     Model = getattr(models, args.model)
     model, _ = Model.load(
         args.path,
         model_checkpoint_file=args.checkpoint_name,
         args=args,
         device=device,
         tasks=args.tasks,
-        src_lang=src_lang,
-        tgt_lang=tgt_lang,
+        src_lang=args.pred_src_languages[0],
```

Comment:
So now `pred_src_languages` and `pred_tgt_languages` are only used for determining the language token of pre-trained models that support it?

Reply:
There are no functional changes. Before you could pass a string of languages (joined by '+') for each task. It was a convenient way of evaluating on multiple languages in one run (e.g. for almond_multilingual) tasks. 
I removed that option since almond_multilingual is gone and the same task can be achieved by running predict multiple times once for each language.